### DISCUSSION TITLE
Show the user where he can download a new version of Jamulus

### DISCUSSION THREAD

**Discussion by ann0see** [jamulussoftware/jamulus#1023] [UI Related]
**Is your feature request related to a problem? Please describe.**

The "new version available message" doesn't explain where you can get the next version of Jamulus from. This might confuse new users

**Describe the solution you'd like**

Either a link to sourceforge or to the upcoming GitHub latest release page

**Describe alternatives you've considered**

A link to the GitHub latest release page in the menu labeled (Get releases (you currently have version x.y.z) ...)

**Comment by gilgongo** [comment]
Maybe we should create a standalone "download" page as per the convention for other projects? That would also mean we could use that link independently of the actual download links on it. Right now we have to take people directly to SourceForge (so we can't "integrate" it into the overall site journey).

**Comment by ann0see** [comment]
Yes. This should be possible with the new release process. We could just use a link to the binary on the installation for Windows/macOS pages

**Comment by ann0see** [comment]
I think this is related to https://github.com/jamulussoftware/jamuluswebsite/issues/307

I'm also not 100% happy about the way I implemented it.

**Comment by pljones** [comment]
Say we have version 3.6.2 with a feature that, when a new version is available, pops up a link.  It's going to pop up `https://sf.net/llcon/download/latest`.  Which isn't where we _really_ want people going -- we want them going to somewhere on jamulus.io.

We're already tied to the central server location for notifying the available upgrade.  I'd rather not tie more locations to versions of the client.  ***NOT EVERYONE INSTALLS EVERY NEW VERSION*** - so you end up constantly tied to old locations with some things redirecting to new ones.

So if we start supplying an upgrade location, it should be _along with_ the notification of the new version: that is, we need to update the protocol to support passing the URL.

**Comment by gilgongo** [comment]
> So if we start supplying an upgrade location, it should be along with the notification of the new version

If I understand correctly, now that we can have https://jamulus.io/download.html - would the URL ever change after that?

**Comment by ann0see** [comment]
The download.html doesn’t exist (yet) but would be easy to create. Currently you can access the direct link to a binary via a Jekyll variable. Yes, this variable will have to be changed on every release

**Comment by gilgongo** [comment]
Sorry, I wasn't being clear, I meant that @pljones says "we need to update the protocol to support passing the URL" but if that URL never changed after that, why bother with a protocol change? I may not be understanding correctly, but it seems unwise to link people directly to a download, since that means we don't have an opportunity to notify them of new things (using a download page).

**Comment by gilgongo** [comment]
**META COMMENT**

I think this is a good example of a conversation that would ideally start out in a discussion area called "Documentation", and then when we have an agreement on what the actual action(s) is, we open a ticket(s) for that in the master issues list (of which there would only be one). Then, if subsequent questions/discussions about that ticket arise, we have them on that ticket, BUT then **update the ticket itself to reflect any changes to it**. That way, if somebody wants to pick up a ticket and do it, they have a clear spec to work to without needing to comb through the discussions on it, as we have to do today.

Minor documentation updates (re-phrasings, corrections, small additions) wouldn't need tickets - they could just be decided in a discussion and then work through in a PR as we do today.

\[END META\]

**Comment by pljones** [comment]
> if that URL never changed after that, why bother with a protocol change?

Two months ago, someone might have said "but Volker will never leave the project".  Things change.

**Comment by chrisrimple** [comment]
Isn't it sufficient to link to jamulus.io and let the user click the Download link (or whatever it might become), until someone decides to implement an auto-upgrader that requires no action by the user (except to approve the upgrade)?

**Comment by gilgongo** [comment]
> Two months ago, someone might have said "but Volker will never leave the project".

:-) Well, that wasn't under our collective control, while _this_ is. 

@chrisrimple 

> Isn't it sufficient to link to jamulus.io and let the user click the Download link

That's what I'm thinking. Pretty much all projects on the web have www.something.tld/download. It's almost a standard now. Why would we not just link everyone to that and be done with it? When the user is on that page, we can have all sorts of detection, dynamic links latest version whatever, no? A protocol change in Jamulus seems a bit heavy to me but, hey.

**Comment by gilgongo** [comment]
I think this may be still under discussion, so moving to there until we have something defined for a ticket.

**Reply by gilgongo** [reply]
Meanwhile, I've raised this to create downloads page: https://github.com/jamulussoftware/jamuluswebsite/issues/322

**Comment by pljones** [comment]
We have an issue related to this: https://github.com/jamulussoftware/jamulus/issues/1178
